### PR TITLE
[WIP] Plans stripped reconstruction

### LINKED ISSUE #5412: Plan stripped re-construction
## Description
Currently, when a plan is created, it contains a lot of metadata like Pointer UIDs, VM addresses, etc. So, if the same plan is created twice, its representation would be completely different.

Solution: to re-construct the plan without any metadata.

## Are you interested in working on this improvement yourself?
- Yes, I am.

## Additional Context
https://github.com/OpenMined/PySyft/issues/5359#issuecomment-807726213

### PR DESCRIPTION
**Pull request by vsquareg**
## Description
Fix #5412

Currently, when a plan is created, it contains a lot of metadata like Pointer UIDs, VM addresses, etc. So, if the same plan is created twice, its representation would be completely different.

This PR attempts to construct a stripped Plan.

**Why?**
- Allows to build a more user-readable source code (https://github.com/OpenMined/PySyft/issues/5359#issuecomment-807726213)
- Make serialization and deserialization deterministic (#5292) i.e. the representation of a plan is solely dependent on the actions that happen inside it and types of pointers it uses.

## Progress
- [ ] Implement all `Action` types [1/10]
- [x] `strip()`
- [ ] Re-construct from `serialized(plan)`

### CODE REVIEW

**Comment by vsquareg** [comment]
Would really appreciate some thoughts and discussions on this. Regarding whether or not this is even required, and if there's a better approach for implementation.

**Comment by koenvanderveen** [comment]
Hey, about this:

> Allows to build a more user-readable source code (#5359 (comment))

Can you explain me how you would like to improve that with the current (or future?) changes?

**Comment by vsquareg** [comment]
> Hey, about this:
> 
> > Allows to build a more user-readable source code (#5359 (comment))
> 
> Can you explain me how you would like to improve that with the current (or future?) changes?

I guess my point was ambiguous, I meant the string representation of the code (say to be sent to the DO for review).
```
@make_plan
def operations(xs = th.rand(5), ys = th.rand(5)):

    ls = xs + ys
    ms = xs * ys

    return ls + ms
```
After stripping the Plan, the above is essentially represented like this:
```
input: xs: inp1, ys: inp2
actions:
- add(inp1, inp2) store in inter1
- mul(inp1, inp2) store in inter2
- add(inter1, inter2) store in inter3
output: inter3
 ```
Although, I suppose it can be done directly from the current code without stripping too.

File: scripts/build_proto.sh [hunk 1]
```diff
@@ -17,11 +17,11 @@ find ${PROTO_IN} -name "*.proto" -print0 | xargs -0 protoc --python_out=${PYTHON
 if [ "$(uname)" == "Darwin" ]; then
     echo "Darwin"
     # note the '' for empty file on MacOS
-    find src/syft/proto -name "*_pb2.py" -print0 | xargs -0 sed -i '' 's/from \(proto.*\) import /from syft.\1 import /g'
```
Comment (madhavajay) [inline scripts/build_proto.sh:20]: Any reason for looking for this files outside of the /proto folder?

Comment (vsquareg) [reply scripts/build_proto.sh]: No, did not intend to push this change.

File: src/syft/core/plan/plan_run_class_method_action.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+from ...proto.core.node.common.action.plan_run_class_method_pb2 import PlanRunClassMethodAction as PlanRunClassMethodAction_PB
+from ..common.serde.deserialize import _deserialize
+from ..common.serde.serializable import bind_protobuf
+# from ..node.common.action.run_class_method_action import RunClassMethodAction
+from .plan_pointer import PlanPointer
+
+
+@bind_protobuf
+class PlanRunClassMethodAction(Serializable):
+    """
```
[the remark this answers is not in the document]
Comment (vsquareg) [reply src/syft/core/plan/plan_run_class_method_action.py:26]: I thought about it earlier, but to me, it seems that `Plans` is dependent on (and uses) `Actions`, and not the other way round.
Making `RunClassMethodAction` inherit from `PlanRunClassMethodAction`, would somehow inherently imply that `Actions` are dependent on `Plans`.

Do you think I should add the inheritance?

**Comment by madhavajay** [comment]
Closing because this is stale. We will revisit this when we understand how we want to handle Plans in the `0.6.0` branch.

**Pull request closed without merging**